compiler/rustc_abi/src/lib.rs

```diff
@@ -1686,6 +1702,11 @@ impl<FieldIdx: Idx, VariantIdx: Idx> LayoutS<FieldIdx, VariantIdx> {
         self.is_sized() && self.size.bytes() == 0 && self.align.abi.bytes() == 1
     }
 
+    /// Returns true if the size of the type is only known at runtime.
+    pub fn is_runtime_sized(&self) -> bool {
+        matches!(self.abi, Abi::ScalableVector { .. })
```

Comment:
This coexisting with `is_unsized` feels potentially misleading, or at the very least ambiguous.

Also, the *possibility* of knowing the size *at all* feels like an edge case out of all possible "value-only"/"memory-unrepresentable" types discussed in e.g. these older RFC comments:
- https://github.com/rust-lang/rfcs/pull/3268#issuecomment-1137781023
- https://github.com/rust-lang/rfcs/pull/3268#issuecomment-1149479054

As far as `Abi` knowing about these types, that's an inevitability (since they do participate in ABI), and sadly I don't know of a good way to encode open-ended target-specific "handle" types (other than e.g. an interned `[Either<Symbol, u64>]` style sequence, comparable to specifying register names using strings in `asm!` etc.).